tests/integration/test_model.py

```diff
@@ -60,6 +60,22 @@ async def test_deploy_local_bundle_file(event_loop):
                           timeout=60 * 4)
 
 
+@base.bootstrapped
+@pytest.mark.asyncio
+async def test_deploy_bundle_local_charms(event_loop):
```

Comment:
I can't see a test for the .charm or .zip scenario?

Reply:
The file is [here in the diff](https://github.com/juju/python-libjuju/pull/470/files#diff-5104c1c0a71e77dc6ad34c63eb0723ae5c4e272c9c5f75aa656eec2292b9e837) and it's referenced [here in the bundle](https://github.com/juju/python-libjuju/pull/470/files#diff-54b5b6ff5ea16d0a1e4ea85dbce3bcc3bc29a7f0e09179804da5787e7892c9c0) which is in turned [used in the test you commented on](https://github.com/juju/python-libjuju/pull/470/files#diff-9405e0bc34a26ab57891171003bb7253b44c20fa921670f44b207c1971b413f1R67).  So it's testing bundle which uses both a local charm dir and a local charm file.